parser/prism/Translator.cc

```diff
@@ -142,6 +142,26 @@ std::unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
                 return sendNode;
             }
         }
+        case PM_CASE_NODE: {
+            auto caseNode = reinterpret_cast<pm_case_node *>(node);
+            pm_location_t *loc = &caseNode->base.location;
+
+            auto predicate = translate(caseNode->predicate);
+
+            auto prismConditions = absl::MakeSpan(caseNode->conditions.nodes, caseNode->conditions.size);
+
+            parser::NodeVec sorbetConditions{};
+            sorbetConditions.reserve(prismConditions.size());
+
+            for (auto &prismCondition : prismConditions) {
+                sorbetConditions.emplace_back(translate(prismCondition));
+            }
```

Comment:
Thought on cleaning some of this up... we could make a method specifically for converting a `pm_node_list` to a `NodeVec`? Like that PR you opened @amomchilov but a dumber implementation.

Reply:
I have a branch locally with exactly this, actually. Merge this, and I'll rebase on top (and apply it here)